tika/src/main/java/fr/pilato/elasticsearch/crawler/fs/tika/TikaInstance.java

```diff
@@ -94,11 +105,62 @@ private static void initParser(Fs fs) {
                         Collections.singletonList(TesseractOCRParser.class));
             }
 
-            Parser PARSERS[] = new Parser[2];
-            PARSERS[0] = defaultParser;
-            PARSERS[1] = pdfParser;
 
-            parser = new AutoDetectParser(PARSERS);
+            // load custom parsers if defined in config
+            if (fs.getCustomTikaParsers().size() > 0) {
+                Integer counter = 0;
+                Parser PARSERS[] = new Parser[fs.getCustomTikaParsers().size()+2];
+
+
+
+                // to collect all Mediatypes handled by custom parser to exclude them form the DefaultParser
+                List<MediaType> excludeMediaTypes = new ArrayList<MediaType>();
+
+
+                for (CustomTikaParser customTikaParser : fs.getCustomTikaParsers()) {
+                    counter += 1;
+                    try {
+                        URL[] jarUrl = { new URL("jar:file:" + customTikaParser.getPathToJar()+"!/") };
+                        URLClassLoader urlClassLoader = URLClassLoader.newInstance(jarUrl);
+                        Class customParserClass = urlClassLoader.loadClass(customTikaParser.getClassName());
+                        Parser customParser = (Parser) customParserClass.newInstance();
+                        String[] customMimeStrings = new String[customTikaParser.getMimeTypes().size()];
+                        Set<MediaType> customMediaTypes = MediaType.set(customTikaParser.getMimeTypes().toArray(customMimeStrings));
+                        excludeMediaTypes.addAll(customMediaTypes);
+                        Parser customParserDecorated = ParserDecorator.withTypes(customParser, customMediaTypes);
+                        PARSERS[counter] = customParserDecorated;
+
+
+
+                    } catch (IOException e) {
```

Comment:
I think you can just do here something like:

```java
} catch (IOException|ClassNotFoundException|InstantiationException|IllegalAccessException e) {
    logger.error("Caught {}: {}", e.getClass().getSimpleName(), e.getMessage());
}
```

Or just catch any `Exception`?